src/Microsoft.CodeQuality.Analyzers/CSharp/QualityGuidelines/CSharpRethrowToPreserveStackDetails.Fixer.cs

```diff
@@ -0,0 +1,33 @@
+﻿using System.Composition;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.Formatting;
+using Microsoft.CodeQuality.Analyzers.QualityGuidelines;
+
+
+namespace Microsoft.CodeQuality.CSharp.Analyzers.QualityGuidelines
+{
+    /// <summary>
+    /// CA2200： Rethrow to preserve stack details
+    /// </summary>
+    [ExportCodeFixProvider(LanguageNames.CSharp), Shared]
+    public sealed class CSharpRethrowToPreserveStackDetailsFixer : RethrowToPreserveStackDetailsFixer
+    {
+        protected async override Task<Document> MakeThrowAsync(Document document, SyntaxNode nodeToReplace, CancellationToken cancellationToken)
```

Comment:
The entire code can be moved to the language agnostic layer by using [SyntaxGenerator.ThrowStatement](http://source.roslyn.io/#Microsoft.CodeAnalysis.Workspaces/Editing/SyntaxGenerator.cs,1430) API to generate the throw statement node - it should work for both C# and VB